### ISSUE TITLE
Version guessing error

### ISSUE THREAD

**Issue by victorgawk** [twmb/franz-go#536]
go version: 1.20.7
franz-go version: 1.14.3
operational system: Windows 10 x64

Kafka version guessing is not working as expected. For a kafka cluster v3.1 it is guessing as `unknown custom version at least v2.7`.

Seems like this bug was introduced at franz-go v1.12.0 (It was working at franz-go v1.11.7).

Code to reproduce:

compose.yaml
```yaml
version: '2.4'

services:

  zookeeper:
    image: bitnami/zookeeper:3.7.0
    container_name: zookeeper
    volumes:
      - zookeeper_data:/bitnami/zookeeper
    environment:
      - ALLOW_ANONYMOUS_LOGIN=yes
      - ZOO_SERVER_ID=0
    networks:
      - kafka_network

  kafka:
    image: bitnami/kafka:3.1.0
    container_name: kafka
    volumes:
      - kafka_data:/bitnami/kafka
    environment:
      - KAFKA_CFG_ZOOKEEPER_CONNECT=zookeeper:2181
      - ALLOW_PLAINTEXT_LISTENER=yes
      - ALLOW_ANONYMOUS_LOGIN=yes
    ports:
      - 9092:9092
    networks:
      - kafka_network
    depends_on:
      - zookeeper

networks:

  kafka_network:
    driver: bridge

volumes:

  zookeeper_data:
    driver: local

  kafka_data:
    driver: local
```

kafka_version_bug.go
```go
package main

import (
	"context"
	"fmt"

	"github.com/twmb/franz-go/pkg/kgo"
	"github.com/twmb/franz-go/pkg/kmsg"
	"github.com/twmb/franz-go/pkg/kversion"
)

func main() {
	seeds := []string{"localhost:9092"}

	cl, err := kgo.NewClient(
		kgo.SeedBrokers(seeds...),
	)
	if err != nil {
		panic(err)
	}
	defer cl.Close()

	ctx := context.Background()

	req := kmsg.NewApiVersionsRequest()
	req.ClientSoftwareVersion = "csVersion"
	req.ClientSoftwareName = "csName"

	apiVersions, err := req.RequestWith(ctx, cl)
	if err != nil {
		panic(err)
	}

	versions := kversion.FromApiVersionsResponse(apiVersions)

	fmt.Println(versions.VersionGuess())
}
```

**Comment by twmb** [comment]
Thanks for the compose, I'll test this today.

**Comment by twmb** [comment]
This has been fixed in v1.14.4
